### ISSUE TITLE
Experimental results

### ISSUE THREAD

**Issue by MaorunZhang** [imirzadeh/Teacher-Assistant-Knowledge-Distillation#13]
Hello, thanks for sharing good codes.

I've just run the code(Example 3.3), using 'resnet110' as teacher, 'resnet20' as TA and 'resnet8' as student on cifar10. 

The accuracy of used resnet110 is 93.17%, then I used the commands in steps 2 and 3(Example 3.3). But the results of training were not ideal. The accuracy of TA(resnet20) is 82.08% and the accuracy of student(resnet8) is 78.01%.

Here is the configuration:
`Namespace(batch_size=128, cuda=True, dataset='cifar10', dataset_dir='./data', epochs=160, learning_rate=0.1, momentum=0.9, student='resnet20', teacher='resnet110', teacher_checkpoint='/home/zmr/experiment/pytorch-classification-original/checkpoints/cifar10/resnet-110/model_best.pth.tar', weight_decay=0.0001)`

Is there anything wrong in my experiment?

Thanks!

**Comment by imirzadeh** [comment]
Hi there,   
 
This issue has already been resolved here: [https://github.com/imirzadeh/Teacher-Assistant-Knowledge-Distillation/issues/10](https://github.com/imirzadeh/Teacher-Assistant-Knowledge-Distillation/issues/10) (more specifically, [here](https://github.com/imirzadeh/Teacher-Assistant-Knowledge-Distillation/issues/10#issuecomment-548648070))  
I suspect the issue is with hyper-parameters. Because as far as I know the accuracies even without knowledge distillation should be much higher.

**Comment by MaorunZhang** [comment]
I‘ve tried and changed the hyper-parameter in search_space.json, but it didn't seem to work.

Then I checked the code. In train.py(function adjust_learning_rate, line 152): 

		if models_are_plane:
			lr = 0.01
		else:
			if epoch < int(epoch/2.0):
				lr = 0.1
			elif epoch < int(epochs*3/4.0):
				lr = 0.1 * 0.1
			else:
				lr = 0.1 * 0.01

There seems to be a bug. I think "if epoch < int(epoch/2.0)" should be "if epoch < int(epochs/2.0)", right?

**Comment by imirzadeh** [comment]
Yes, you are right. I kind of cleaned the code when I made this repository. I'm sorry for the typo.  However, I still have the "messy" code and the results logged here:   
This is the link to the experiment of distilling from resnet 110 to resnet20.  [https://www.comet.ml/cifar10/c10res110res20/11d9c1d561aa4afbabcc52fc1a04962e?experiment-tab=chart](https://www.comet.ml/cifar10/c10res110res20/11d9c1d561aa4afbabcc52fc1a04962e?experiment-tab=chart)      

You can see the **code** here: [https://www.comet.ml/cifar10/c10res110res20/11d9c1d561aa4afbabcc52fc1a04962e?experiment-tab=code](https://www.comet.ml/cifar10/c10res110res20/11d9c1d561aa4afbabcc52fc1a04962e?experiment-tab=code)  
and **hyperparameters** here:   [https://www.comet.ml/cifar10/c10res110res20/11d9c1d561aa4afbabcc52fc1a04962e?experiment-tab=params](https://www.comet.ml/cifar10/c10res110res20/11d9c1d561aa4afbabcc52fc1a04962e?experiment-tab=params)  

You can see that the resnet20 gets the best accuracy of 92.82  
I hope it helps.

**Comment by MaorunZhang** [comment]
Thank you very much for the original code. 

I'd like to replicate your experiments,  so I ran some experiments on this code. I used a different model as a teacher(the accuracy of my resnet110 is 93.17%), which may cause my experiment results to be different from yours. More precisely, my results are a little lower than yours. 

So, I wonder if I can use this code to train a resnet110 like yours: [https://www.comet.ml/cifar10/c10res110/0bfabfce4eee4979be7e596f832f966f?experiment-tab=code](url)

Or, if it's convenient for you, can you provide me with your trained resnet110?

**Comment by imirzadeh** [comment]
The weight file is on assets tab of the link you had.
You can use this too:
https://www.comet.ml/api/asset/download?experimentKey=0bfabfce4eee4979be7e596f832f966f&assetId=5b9c7f0543d44c5cb817d2ebfa0941a2&isCompressed=false
